qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -291,23 +292,25 @@ def __init__(
                 use_full_support=False,
             )
             hmat = hmat.permute(1, 2, 0)
-            self._hamiltonian = lambda x: hmat
+            self.register_buffer("hmat", hmat)
+            self._hamiltonian = lambda self, values: self.hmat
 
         elif isinstance(block.generator, sympy.Basic):
             self._hamiltonian = (
-                lambda values: values[self.param_names[1]].squeeze(3).permute(1, 2, 0)
+                lambda self, values: values[self.param_names[1]].squeeze(3).permute(1, 2, 0)
             )
             # FIXME Why are we squeezing
         else:
 
-            def _hamiltonian(values: dict[str, Tensor]) -> Tensor:
+            def _hamiltonian(self: PyQHamiltonianEvolution, values: dict[str, Tensor]) -> Tensor:
+                _dev = list(values.values())[0].device
```

Comment:
Could an empty dict be passed here? For unparameterized Hamiltonian?

Reply:
not really, this should only be executed if you have parametrized generators although its not really clear from this long __init__.  non-parametric generators will go into the first if clause